### ISSUE TITLE
namespaced event adds unnamespaced method to object instance

### ISSUE THREAD

**Issue by pickerflicker** [aasm/aasm#519]
Unsure if this is a bug or expected behavior.

```
class Order
  aasm(:state, name_space: :billing) do
    event :charge do ... end
  end
end
```

This adds both namespaced and unnamespaced methods to the Order object:

``` [:charge, :charge_billing]```

I would expect only `:charge_billing` to show up. Was hitting unexpected method name collisions as a result.

**Comment by romankovt** [comment]
and also it produces warnings for overriding method if you have the same event names but in different namespaces

**Comment by tfwright** [comment]
I agree this is unexpected and severely reduces the usefulness of namespaces, which would seem to be precisely to avoid this kind of collision. The current implementation makes namespaces essentially cosmetic because you can't actually use the desired event names because you need to add the namespace yourself.

From reading through [this thread](https://github.com/aasm/aasm/issues/469) it appears to be an intentional decision to avoid a breaking change, but I think this merits a breaking change and would vote to refactor that code to change the method definition when namespaces are used rather than aliasing them.

**Comment by euanlau** [comment]
@anilmaurya what's your take on this?

**Comment by anilmaurya** [comment]
@euanlau I agree with @tfwright . As this is a breaking change, it will be fixed in `AASM 6`

**Comment by afuno** [comment]
@alto @anilmaurya Guys, this problem has been going on for several years now. Is there any estimate on how it's been resolved?

**Comment by alto** [comment]
This seems to be a bit of technical debt left behind from https://github.com/aasm/aasm/issues/469 . Path forward would be clean this up which would be a breaking change and therefore it has been been labeled with "aasm6". I can start looking into a v6 with all those [issues and PRs](https://github.com/aasm/aasm/issues?q=state%3Aopen%20label%3Aaasm6).

**Comment by alto** [comment]
@pickerflicker Do you mind taking a look at https://github.com/aasm/aasm/pull/880 for me please? This should fix the method bleeding.
